frontend/src/metabase/visualizations/shared/components/RowChartView/utils/data-labels.unit.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { scaleLinear } from "d3-scale";
+
+import type { BarData } from "../../RowChart/types";
+
+import { getDataLabel } from "./data-labels";
+
+describe("getDataLabel", () => {
+  const mockXScale = scaleLinear().domain([0, 100]).range([0, 500]);
+  const seriesKey = "test-series";
+
+  const createBarData = (overrides: Partial<BarData<any>> = {}): BarData<any> =>
```

Comment:
I'm not a huge fan of `any`. I get why it's used on L18, but maybe we can use a generic in this create function?
```suggestion
  const createBarData = <T>(overrides: Partial<BarData<T>> = {}): BarData<T> =>
```